flume-hdfs-sink/src/main/java/com/wipro/ats/bdre/flume/sink/hdfs/BucketWriter.java

```diff
@@ -413,13 +404,12 @@ public synchronized void flush() throws IOException, InterruptedException {
     if (!isBatchComplete()) {
       doFlush();
 
-      if(idleTimeout > 0) {
-        // if the future exists and couldn't be cancelled, that would mean it has already run
-        // or been cancelled
-        if(idleFuture == null || idleFuture.cancel(false)) {
+      // if the future exists and couldn't be cancelled, that would mean it has already run
+      // or been cancelled
+      if(idleTimeout > 0  && (idleFuture == null || idleFuture.cancel(false))) {
           Callable<Void> idleAction = new Callable<Void>() {
             public Void call() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1161)